pkg/engine/internal/planner/logical/builder.go

```diff
@@ -48,6 +49,38 @@ func (b *Builder) Parse(kind ParserKind) *Builder {
 	}
 }
 
+// Unwrap applies an [Projection] operation, with an [UnaryOp] unwrap operation, to the Builder.
+func (b *Builder) Unwrap(identifier string, operation string) *Builder {
```

Comment:
Hm, I intended Builder to have an API which more closely reflects the logical plan rather than LogQL constructs. 

Can we update this to `Builder.Cast` maybe?

Reply:
oh, interesting, I though of builder as the bridge between logql and the logical plan, so the parameters of this function match closely to what we get out of logql for the `unwrap` operation. I'm happy to go with `Cast` though, as I think the arguments are still consistent.